src/library/scala/collection/immutable/NumericRange.scala

```diff
@@ -287,6 +300,9 @@ object NumericRange {
       }
       // If we reach this point, deferring to Int failed.
       // Numbers may be big.
+      if (num.isInstanceOf[Numeric.BigDecimalAsIfIntegral]) {
+        bigDecimalCheckUnderflow(start, end, step)  // Throw exception if math is inaccurate (including no progress at all)
```

Comment:
The checks don't look like they would help with multiplication. Does it make sense to still demand precise steps (without rounding errors)?

Reply:
@szeiger - My impression, looking through the math, was that if the addition was precise, the multiplication would be also.  I don't have a proof, unfortunately, or even a good test.

Reply:
Also, if we _don't_ have precise steps, then `BigDecimalAsIfIntegral` is not actually true.  (This also suggests that `BigDecimalAsIfIntegral` should do its math (at least addition/subtraction) at unlimited precision; I could do that instead.  That would be possibly better solution, but one requiring more effort.)

Reply:
I think you're right. I wasn't able to come up with a case where the tested invariants would hold but the calculations of length and individual steps would be inconsistent.

Reply:
I need to check stuff like -6 to 6 by 0.1 with precision 2, but I think the algorithm already is paranoid about avoiding zero, so it should be okay.  (I will check, though.)

Reply:
New test covers this case.